packages/components/src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,107 @@
+import * as DialogPrimitive from '@radix-ui/react-alert-dialog';
+import { ComponentPropsWithoutRef, ElementRef, forwardRef } from 'react';
+
+import { cn } from '~/lib/utils';
+
+const Dialog = DialogPrimitive.Root;
+
+const DialogTrigger = forwardRef<
+  ElementRef<typeof DialogPrimitive.Trigger>,
+  ComponentPropsWithoutRef<typeof DialogPrimitive.Trigger>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Trigger className={cn('', className)} ref={ref} {...props} />
+));
+
+DialogTrigger.displayName = DialogPrimitive.Trigger.displayName;
+
+const DialogPortal = forwardRef<
+  ElementRef<'div'>,
```

Comment:
```suggestion
  ElementRef<typeof DialogPrimitive.Portal>,
```

Reply:
when I tried to provide `ref` to `Portal` I got an error:
![Screenshot 2024-04-16 at 09 43 07](https://github.com/bigcommerce/catalyst/assets/67792608/702e7d1f-f6a1-4e12-9a80-04a0c5dfc2dc)

So I wrapped `Portal` into `<div/>` and put `ref` on `div`...

Reply:
Ahh, got it!